src/array_serialize.rs

```diff
@@ -74,7 +74,10 @@ impl<A, S, D> Decodable for ArrayBase<S, D>
             let elements = try!(
                 d.read_struct_field("data", 2, |d| {
                     d.read_seq(|d, len| {
-                        if len != dim.size() {
+                        let dim_size = dim
+                            .size_checked()
+                            .ok_or_else(|| d.error("overflow calculating array size"))?;
+                        if len != dim_size {
```

Comment:
Nice! This reminds me, it's more than time to drop rustc-serialize I think :)

Reply:
Yeah, I was thinking the same thing. :)